File: test/FunctionTester.vb
<TestFixture()> _
Public Class FunctionTester

    <Test()> Public Sub TestScript()
        Dim f As New Routine("dbo", "udf_GetDate")
        f.Text = _
        "CREATE FUNCTION [dbo].[udf_GetDate]()" + vbCrLf _
        + "RETURNS DATETIME AS" + vbCrLf _
        + "BEGIN" + vbCrLf _
        + "   RETURN GETDATE()" + vbCrLf _
        + "END" + vbCrLf

        Console.WriteLine(f.ScriptCreate())
        TestHelper.ExecSql(f.ScriptCreate())
        TestHelper.ExecSql("drop function [dbo].[udf_GetDate]")
    End Sub

End Class

File: test/FunctionTester.cs
using System;
using System.Linq;
using NUnit.Framework;
using SchemaZen.Library.Models;

namespace SchemaZen.Tests {
	[TestFixture]
	public class FunctionTester {
		private const string _exampleFunc = @"
CREATE FUNCTION [dbo].udf_GetDate()
RETURNS DATETIME AS
BEGIN
	RETURN GETDATE()
END
";

		[Test]
		public void TestScript() {
			var f = new Routine("dbo", "udf_GetDate", null) {
				RoutineType = Routine.RoutineKind.Function,
				Text = _exampleFunc
			};
			Console.WriteLine(f.ScriptCreate());
			TestHelper.ExecBatchSql(f.ScriptCreate() + "\nGO", "");
			TestHelper.ExecSql("drop function [dbo].[udf_GetDate]", "");
		}

		[Test]
		public void TestScriptNoWarnings() {
			var f = new Routine("dbo", "udf_GetDate", null) {
				Text = _exampleFunc,
				RoutineType = Routine.RoutineKind.Function
			};
			Assert.IsFalse(f.Warnings().Any());
		}

		[Test]
		public void TestScriptWarnings() {
			var f = new Routine("dbo", "udf_GetDate2", null) {
				Text = _exampleFunc,
				RoutineType = Routine.RoutineKind.Function
			};
			Assert.IsTrue(f.Warnings().Any());
		}
	}
}
